### DISCUSSION TITLE
Feature Request: Get schema names used in a query

### DISCUSSION THREAD

**Discussion by akshayka** [duckdb/duckdb#13621] [Ideas]
Why do you want this feature?

I'm building an open-source reactive notebook for SQL (and Python) called [marimo](https://github.com/marimo-team/marimo), and our support for SQL is powered by duckdb. 

For each SQL query, we need to determine the schemas the query attaches, and the schemas the query references. The former is readily accomplished by tokenizing the query and checking if it's an `ATTACH` statement. But it's not obvious to me how to find all the schemas that an arbitrary query references.

It would be very helpful if there were a library function in the Python client of the form

```python
duckdb.get_schema_names(query)
```

analogous to `duckdb.get_table_names(query)`.

For a simple example, for the query

```sql
SELECT * FROM chinook.tracks
```

`duckdb.get_schema_names(query)` would return `chinook`. 

As an alternative, it would be even better if `get_table_names` could return a fully-qualified table name, including the schema, as proposed in https://github.com/duckdb/duckdb/discussions/11019.

Any guidance would be appreciated. Thank you!

**Comment by Tishj** [comment]
I also think that it would be better for `get_table_names` to return fully qualified table names, adding the schema + catalog

**Reply by Tishj** [reply]
The way GetTableNames is implemented is with a `BindingMode`, the `BindingMode::EXTRACT_NAMES` mode
Searching for this in the codebase should help you find where this is being applied.

**Comment by taniabogatsch** [comment]
https://github.com/duckdb/duckdb/discussions/11019#discussioncomment-12071044
